Added documentation for new repo datadog-custom-rules-template (#35921)
DataDog/documentation: content/en/security/code_security/static_analysis/custom_rules/_index.md

A ruleset must have a unique name with only letters, numbers, and hyphens (`-`). Examples of valid
ruleset names are `python-security`, `cobra-team-checks`, or `my-company-security-checks`.

## Manage rules with the template repository

> As an alternative to managing custom rules in Datadog, the [datadog-custom-rules-template][5] GitHub repository provides a Git-based workflow for managing custom rules as code. On every push to `main`, a GitHub Action automatically creates, updates, or deletes rulesets and rules to match what's on disk.

Reviewer: ```suggestion
As an alternative to managing custom rules in Datadog, the [datadog-custom-rules-template][5] GitHub repository provides a Git-based workflow for managing custom rules as code. On every push to `main`, a GitHub Action automatically creates, updates, or deletes rulesets and rules to match what's in the repository.
```

### Get started

1. Click **Use this template** on the [datadog-custom-rules-template][5] repository to create your own copy.
2. Add your Datadog credentials as GitHub secrets. See [Authentication](#authentication).
3. Rename `rulesets/my-custom-rules/` or add new ruleset directories under `rulesets/`.
4. Push to `main`. The GitHub Action uploads your rules automatically.

### Authentication

1. In your GitHub repository, go to **Settings → Secrets and variables → Actions**.
2. Add three secrets:
   - `DD_API_KEY` — your Datadog API key
   - `DD_APP_KEY` — your Datadog Application key
   - `DD_SITE` — your Datadog site hostname (for example, `datadoghq.com`, `datadoghq.eu`, or `us3.datadoghq.com`)

### How sync works

On every push to `main`, the GitHub Action runs `upload.py`, which:

> - **Creates** rulesets and rules that are new on disk

Reviewer: ```suggestion
- **Creates** rulesets and rules that are new in the repository
```

- **Updates** rulesets and rules whose content has changed

> - **Deletes** rulesets and rules that have been removed from disk

Reviewer: ```suggestion
- **Deletes** rulesets and rules that have been removed the repository
```

Only changed rules trigger API calls—unchanged rules are skipped.

> To trigger a sync without pushing a commit, go to the **Actions** tab and click **Upload Custom Rules → Run workflow**.

Reviewer: ```suggestion
To run a sync manually, in to your GitHub repository, go to the **Actions** tab, select **Upload Custom Rules**, and click **Run workflow**.
```
